gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/java/io/gravitee/repository/jdbc/management/JdbcApiKeyRepository.java

```diff
@@ -163,33 +237,137 @@ private boolean addClause(boolean first, StringBuilder query) {
-                .query(
-                    getOrm().getSelectAllSql() + " where " + escapeReservedWord("key") + " = ? and api = ?",
-                    getOrm().getRowMapper(),
-                    key,
-                    api
-                )
-                .stream()
-                .findFirst();
+            String query =
+                getOrm().getSelectAllSql() +
+                " k" +
+                " join " +
+                KEY_SUBSCRIPTION +
+                " ks on ks.key_id = k.id" +
+                " join " +
+                SUBSCRIPTION +
+                " s on ks.subscription_id = s.id " +
+                " where k.key = ?" +
+                " and s.api = ?";
+
+            CollatingRowMapper<ApiKey> rowMapper = new CollatingRowMapper<>(getOrm().getRowMapper(), CHILD_ADDER, "id");
+
+            jdbcTemplate.query(query, rowMapper, key, api);
+
+            return rowMapper.getRows().stream().findFirst();
         } catch (final Exception ex) {
             LOGGER.error("Failed to find api key by key and api", ex);
             throw new TechnicalException("Failed to find api key by key and api", ex);
         }
     }
+
+    @Override
+    public Optional<ApiKey> findById(String id) throws TechnicalException {
+        LOGGER.debug("JdbcApiKeyRepository.findById({})", id);
+        try {
+            String query =
+                getOrm().getSelectAllSql() + " k " + " left join " + KEY_SUBSCRIPTION + " ks on ks.key_id = k.id" + " where k.id = ?";
+
+            CollatingRowMapper<ApiKey> rowMapper = new CollatingRowMapper<>(getOrm().getRowMapper(), CHILD_ADDER, "id");
+
+            jdbcTemplate.query(query, rowMapper, id);
+
+            return rowMapper.getRows().stream().findFirst();
+        } catch (final Exception ex) {
+            LOGGER.error("Failed to find key by id", ex);
+            throw new TechnicalException("Failed to find key by id", ex);
+        }
+    }
+
+    @Override
+    public Set<ApiKey> findAll() throws TechnicalException {
+        LOGGER.debug("JdbcApiKeyRepository.findAll()");
+        try {
+            String query =
+                getOrm().getSelectAllSql() +
+                " k" +
+                " join " +
+                KEY_SUBSCRIPTION +
+                " ks on ks.key_id = k.id" +
+                " order by k.updated_at desc ";
+
+            CollatingRowMapper<ApiKey> rowMapper = new CollatingRowMapper<>(getOrm().getRowMapper(), CHILD_ADDER, "id");
+
+            jdbcTemplate.query(query, rowMapper);
+
+            return new HashSet<>(rowMapper.getRows());
+        } catch (final Exception ex) {
+            LOGGER.error("Failed to find all keys", ex);
+            throw new TechnicalException("Failed to find all keys", ex);
+        }
+    }
+
+    private void storeSubscriptions(ApiKey key) {
+        List<String> subscriptions = key.getSubscriptions();
+
+        jdbcTemplate.update("delete from " + KEY_SUBSCRIPTION + " where key_id = ?", key.getId());
```

Comment:
Not really related to this line of code, but..

We can delete subscriptions, using SubscriptionService.delete.
(Which actually not deletes Api keys).

I think that will lead to a broke FK in key_subscription ?
Maybe we have to add some key_subscription deletion during the subscription deletion process.